dp/decompose_flow_vectors.py

```diff
@@ -16,6 +16,63 @@
 logger.setLevel(logging.DEBUG)  # For testing, overridden by -l when run as a script
 
 
+def check_and_raise_exception(condition, source, err_msg, exc):
+    if condition:
+        logger.critical(err_msg)
+        source.close()
+        raise exc(err_msg)
+
+
+def dimensions_validity(variable):
+    return (
+        hasattr(variable, "dimensions")
+        and "lon" in variable.dimensions
+        and "lat" in variable.dimensions
+    )
+
+
+def source_check(source):
+    source_file_path = source.filepath()
+
+    condition = not dimensions_validity(source)
+    err_msg = "{} does not have latitude and longitude dimensions".format(
+        source_file_path
+    )
+    check_and_raise_exception(condition, source, err_msg, AttributeError)
```

Comment:
While I appreciate the cleverness of what you have done here, I _do_ think an `if` statement is more understandable. The way it is written right now we are _always_ creating an error message and _then_ checking if there is an error. I can imagine this being challenging to follow in the future.

Reply:
I had to do this because codeclimate did not like similar blocks repeating in a file. May I ignore what codeclimate suggests?

Reply:
> May I ignore what codeclimate suggests?

_Probably_ not. What specifically was it complaining about? Just the fact that we had two similar `if` statements that would build an `err_msg` then handle the exception?

Reply:
It was because of these if statements:
[R28-R32](https://github.com/pacificclimate/climate-explorer-data-prep/pull/127/commits/93a5200a12b22c6cf0bf1f63a842ec8c7e22a58f#diff-1e7f88c622597ae3ca67157373b72288R28-R32)
[R59-R61](https://github.com/pacificclimate/climate-explorer-data-prep/pull/127/commits/93a5200a12b22c6cf0bf1f63a842ec8c7e22a58f#diff-1e7f88c622597ae3ca67157373b72288R59-R61)

Reply:
Ok, and what exactly was the complaint from codeclimate?

Reply:
Ok, I can see what it's complaining about.  That is a little bit annoying haha,  I think we _could_ ignore the error. We may be able to combine [these lines](https://github.com/pacificclimate/climate-explorer-data-prep/pull/127/commits/93a5200a12b22c6cf0bf1f63a842ec8c7e22a58f#diff-1e7f88c622597ae3ca67157373b72288R59-R65) into an `if/elif` statement, but I'm not sure if that would be enough. Maybe give that a try?